ludumdare/ludumdare
# Notifications settings cookie

Uses cookie not tied to the user in any way but shared for the computer for storing notifications settings. It makes communication between the navigation bar and the feed possible and keeping settings between sessions. Logic for this is in `shrub/js/notification/notifications.js` with the intent that if we move it to server-side settings, the components could stay as they are.

It adds a new notification type `feedback` for all notes on a post you're an author to. 

It only shows notification count for the types you're interested in.

It causes the notification count to be correct up to 40 new notifications given that all notifications you have are kept (not filtered out). If there are more than 40 new notifications, you'll get a number that may be lower than the actual number of notifications. It doesn't show that there may be more but code for doing such is included, but commented out.

If you get new notifications while the dropdown is expanded or you update your settings in the feed it will also update on next poll. 

If you mark notifications as read in the dropdown and there were more hidden notifications, they will show next poll. All filtered out notifications in the batch of 40 that were actually polled will be marked as read.

@sgstair Please check the changes in the php-code...

## Tests

* Make
* Run simulator several times
* Use the UI and behaves as expected.
* It can handle and render notifications of unknown type, but they will have very weird wording.
